### DISCUSSION TITLE
How to get True or False status of a checkbox

### DISCUSSION THREAD

**Discussion by saqwild** [writer/writer-framework#178] [Q&A] [answered]
Hello,
As I am working with checkbox, I would appreciate two clarifications.
1/ How to mark a checkbox "checked" by default? (unless I missed, I did not noticed this on documentation page)
2/ is there any event sends the True of False status of a checkbox?

Currently I am working on a plotly graph, the x-axis of the graph is linked with a checkbox. (Since there is no toggle button available, I have to rely on the check box for this)
When the checkbox is not checked the x-axis of the graph is based on the "years" otherwise "months", however I am not able to get the  correct status of the click.
I know there is an "on_change() function but that returns the payload which is "True" by default.

in my code i have defined and linked the CB with state variable like this.
```python
 "CBYearOrYear": {
            "byYr": "Years",
        },
``` 
and this reflect to the grph x-axis.

```python
#this is always true
 typeXAxis = "Years" if CBYearOrYear["byYr"] else "Months"

    fig = px.histogram(
        main_df,
        x=typeXAxis,
......
``` 
I also tried to toggle the checkbox with on_change() and getting the payload, but it gives weird results.

Appreciate your assistance.

**Comment by ramedina86** [comment, marked as answer]
Hi,

In Streamsync, the value of a _Checkbox Input_ is an array, since it can contain multiple options, and multiple options can be selected at once (unlike with a radio input, where only one option can be selected).

For example, if you have a _Checkbox Input_ with three options (keys _a_, _b_, _c_), you can set _a_ and _b_ to be checked by binding a state element as shown below. 

```py
initial_state = ss.init_state({
    "my_checks": ["a", "b"]
})
```

<img width="795" alt="Screenshot 2023-11-25 at 09 29 56" src="https://github.com/streamsync-cloud/streamsync/assets/64783088/0248cc62-a58c-4685-b6af-0b5ce5d2eb2d">

Hope that helps!

**Reply by saqwild** [reply]
Thank you [ramedina86](https://github.com/ramedina86) for your help, yes that helped me to understand the logic and advance further, thought I had to create a temp state and assign the chart x-axix to it and then linked with appropriate chart param.
I could have saved some bites of code if i was able to access the payload outside the on_change state. Appreciate if you could also review [this](https://github.com/streamsync-cloud/streamsync/discussions/169) and confirm accordingly.
